plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -201,11 +202,17 @@ void CppParser::addCompileCommand(
 
     model::BuildSource buildSource;
     buildSource.file = _ctx.srcMgr.getFile(srcTarget.first);
+    buildSource.file->parseStatus = error_
+      ? model::File::PSPartiallyParsed
+      : model::File::PSFullyParsed;
     buildSource.action = buildAction;
     sources.push_back(std::move(buildSource));
 
     model::BuildTarget buildTarget;
     buildTarget.file = _ctx.srcMgr.getFile(srcTarget.second);
+    buildTarget.file->parseStatus = error_
```

Comment:
The build target is an object file or an executable. These are not parsed, so these shouldn't be partially or fully parsed. These are not parsed and this is the default value of parse status. So you don't have to set it at all.